src/shared/watchdog.c

```diff
@@ -29,6 +30,40 @@ static unsigned saturated_usec_to_sec(usec_t val) {
         return (unsigned)(t >= WATCHDOG_TIMEOUT_MAX_SEC ? WATCHDOG_TIMEOUT_MAX_SEC : t); /* Saturate to watchdog max */
 }
 
+static int update_pretimeout(void) {
+        if (watchdog_fd < 0)
+                return 0;
+
+        if (watchdog_pretimeout == USEC_INFINITY || watchdog_timeout == USEC_INFINITY)
+                return 0;
+        else if (saturated_usec_to_sec(watchdog_pretimeout) >= saturated_usec_to_sec(watchdog_timeout)) {
+                unsigned t_sec = saturated_usec_to_sec(watchdog_timeout);
+                unsigned pt_sec = saturated_usec_to_sec(watchdog_pretimeout);
+                log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "Cannot set watchdog pretimeout to %us (%s watchdog timeout of %us)",
+                                pt_sec, pt_sec == t_sec ? "same as" : "longer than", t_sec);
+                return -EINVAL;
+        } else {
+                unsigned sec = saturated_usec_to_sec(watchdog_pretimeout);
+
+                if (ioctl(watchdog_fd, WDIOC_SETPRETIMEOUT, &sec)) {
+                        /* EOPNOTSUPP means the watchdog does not support pretimeouts */
+                        log_full_errno(ERRNO_IS_NOT_SUPPORTED(errno) ? LOG_INFO : LOG_WARNING, errno,
+                                       "Failed to set pretimeout to %us: %m", sec);
+                        return ERRNO_IS_NOT_SUPPORTED(errno) ? 0 : -errno;
+                }
+
+                /* The set ioctl does not return the actual value set, get it now */
+                if (ioctl(watchdog_fd, WDIOC_GETPRETIMEOUT, &sec))
+                        log_warning_errno(errno, "Failed to get pretimeout value: %m, ignoring.");
```

Comment:
we usually write that as:

```
Failed to get pre-timeout value, ignoring: %m
```
i.e. we put the `, ignoring` before the %m, not after.